Sources/Descriptor/single_bond.cpp

```diff
@@ -0,0 +1,117 @@
+#include "Descriptor/single_bond.h"
+#include "Descriptor/radial.h"
+#include "Descriptor/y_grad.h"
+#include "Environment/local_environment.h"
+#include <cmath>
+#include <iostream>
+using namespace std;
+
+void single_bond_update_env(
+    Eigen::VectorXd &single_bond_vals, Eigen::MatrixXd &force_dervs,
+    Eigen::MatrixXd &stress_dervs,
+    void (*basis_function)(double *, double *, double, int, vector<double>),
+    void (*cutoff_function)(double *, double, double, vector<double>), double x,
+    double y, double z, double r, int s, int environoment_index,
+    int central_index, double rcut, int N, int lmax,
+    const vector<double> &radial_hyps, const vector<double> &cutoff_hyps) {
+
+  // Calculate radial basis values.
+  double *g = new double[N];
```

Comment:
for arrays that only exist in a function, please consider using std:vector, instead of dynamically assigned pointer